File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        n = len(s)
        
        if k > n:
            return False
        
        mySet = set()
        for i in range(n - k + 1):
            mySet.add(s[i: i + k])
            
        return len(mySet) == (1 << k)

File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        return solve(s, k);
    }
private:
    bool solve(string s, int k) {
        int n = s.length();
        
        if(k > n)
            return false;
        
        unordered_set<string> set;
        for(int i = 0; i < n - k + 1; ++i) {
            set.insert(s.substr(i, k));
        }
        return set.size() == (1 << k);
    }
};